bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpServiceImpl.java

```diff
@@ -64,35 +70,42 @@
 
 @NonNullByDefault
 @Component(configurationPid = "binding.snmp", service = SnmpService.class)
-public class SnmpServiceImpl implements SnmpService {
+public class SnmpServiceImpl implements SnmpService, NetworkAddressChangeListener {
     private final Logger logger = LoggerFactory.getLogger(SnmpServiceImpl.class);
 
     private @Nullable Snmp snmp;
     private @Nullable DefaultUdpTransportMapping transport;
+    private final NetworkAddressService networkAddressService;
 
     private final List<CommandResponder> listeners = new ArrayList<>();
     private final Set<UserEntry> userEntries = new HashSet<>();
+    private Map<String, Object> config = new HashMap<>();
 
     @Activate
-    public SnmpServiceImpl(Map<String, Object> config) {
+    public SnmpServiceImpl(Map<String, Object> config, @Reference NetworkAddressService networkAddressService) {
         addProtocols();
         OctetString localEngineId = new OctetString(MPv3.createLocalEngineID());
         USM usm = new USM(SecurityProtocols.getInstance(), localEngineId, 0);
         SecurityModels.getInstance().addSecurityModel(usm);
-
+        this.networkAddressService = networkAddressService;
+        networkAddressService.addNetworkAddressChangeListener(this);
         modified(config);
+        this.config = config;
```

Comment:
This should not be necessary when `modified` handles the configuration.

Reply:
Fixed.